src/Datadog.Trace/SpanContext.cs

```diff
@@ -78,6 +84,11 @@ private SpanContext(ulong? traceId, string serviceName)
         /// </summary>
         public string ServiceName { get; set; }
 
+        /// <summary>
+        /// Gets the origin of the trace
+        /// </summary>
+        internal string Origin { get; }
```

Comment:
I think this would be worth adding to the `ISpanContext` interface. That would also remove the type-check you currently have in the constructor

Reply:
The specification for this feature as @kevingosse spotted in a previous comment is: `_dd.origin is a special tag that is used then in our backend to change the sampling logic. This tag is not visible in our UI.`

That's the reason to keep it internal, do we want to expose this in the api?

Reply:
Whoops, I didn't see Kevin's comment. Now that I have a better understanding, no I don't think we should expose this property publicly. Disregard my initial comment 😄